ops/testing.py

```diff
@@ -1044,8 +1053,22 @@ def __init__(self, unit_name, meta):
         # {socket_path : _TestingPebbleClient}
         # socket_path = '/charm/containers/{container_name}/pebble.socket'
         self._pebble_clients = {}  # type: {str: _TestingPebbleClient}
+        self._pebble_clients_can_connect = {}  # type: {_TestingPebbleClient: bool}
         self._planned_units = None
 
+    def _can_connect(self, pebble_client) -> bool:
+        return self._pebble_clients_can_connect[pebble_client]
+
+    def _set_can_connect(self, pebble_client, val):
+        if pebble_client not in self._pebble_clients_can_connect:
+            msg = 'cannot set can_connect for the client - are you running a "real" pebble test?'
```

Comment:
Do we want to just silently skip this operation in the cases where someone is running a real pebble test?

Real vs. fake pebble tests are distinguished by an environment variable, which means that folks are likely to use the same test suite for both, correct?

Reply:
The env var actually just affects testing code - nothing in the harness/framework.  I did look into this a bit recently - the idea of making it easy for charms to have their own real pebble tests, but currently, there really isn't any "working with both" code without explicitly writing pluggable fixture stuff like is in test_testing.py or test_pebble.py.  I think the hard error is probably good for now and we can relax it going forward if we start accomodatingcharm real pebble testing more explicitly.

Reply:
I've made some doc string improvements.